fuse_loss/test/test_scaled_loss.cpp

```diff
@@ -0,0 +1,215 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2020, Clearpath Robotics
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <fuse_core/serialization.h>
+#include <fuse_loss/huber_loss.h>
+#include <fuse_loss/scaled_loss.h>
+
+#include <ceres/autodiff_cost_function.h>
+#include <ceres/problem.h>
+#include <ceres/solver.h>
+#include <gtest/gtest.h>
+
+TEST(ScaledLoss, Constructor)
+{
+  // Create a default loss
+  {
+    fuse_loss::ScaledLoss scaled_loss;
+    EXPECT_EQ(1.0, scaled_loss.a());
+    EXPECT_EQ(nullptr, scaled_loss.loss());
+  }
+
+  // Create a loss with a parameter
+  {
+    const double a{ 0.3 };
+    fuse_loss::ScaledLoss scaled_loss(a);
+    EXPECT_EQ(a, scaled_loss.a());
+    EXPECT_EQ(nullptr, scaled_loss.loss());
+  }
+
+  // Create a loss with a parameter and loss function to scale
+  {
+    std::shared_ptr<fuse_loss::HuberLoss> loss{ new fuse_loss::HuberLoss };
+
+    const double a{ 0.3 };
+    fuse_loss::ScaledLoss scaled_loss(a, loss);
+    EXPECT_EQ(a, scaled_loss.a());
+    EXPECT_NE(nullptr, scaled_loss.loss());
```

Comment:
The scaled loss function is sharing ownership with the input parameter. Can't we test that both have the same internal raw pointer?
```
EXPECT_EQ(loss.get(), scaled_loss.loss().get())
```